### PR TITLE
[_]: Feat/send notification on subscription changed

### PR DESCRIPTION
**Pull request by fabioespinosa**
Sends a `post` request to notification service once a subscription has changed.

**Note: Before merge these env variables must be provided:**
- `NOTIFICATIONS_URL`
- `NOTIFICATIONS_API_KEY`

### CODE REVIEW

File: src/services/NotificationService.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    try {
+      const res = await axios.post(process.env.NOTIFICATIONS_URL as string, data, {
+        headers: { 'X-API-KEY': process.env.NOTIFICATIONS_API_KEY as string },
+      });
+      if (res.status !== 201)
+        console.warn(
+          `Post to notifications service failed with status ${res.status}. Data: ${JSON.stringify(data, null, 2)}`,
+        );
+    } catch (err) {
+      console.warn(
```
Comment (fabioespinosa) [inline src/services/NotificationService.ts:46]: couldn't find a better way to use fastify.log here. Perhaps we could export the Fastify instance as a singleton from `app.ts`

Defaulted to console.

File: src/webhooks/index.ts [hunk 2]
```diff
@@ -45,6 +45,7 @@ export default function (
           await handleSubscriptionCanceled(
```
Comment (fabioespinosa) [inline src/webhooks/index.ts:45]: Perhaps this switch has now grown big enough to have an event bus who handles these events (just a suggestion)

Comment (sg-gs) [reply src/webhooks/index.ts:45]: I would say a map serialized or an event bus could be useful, but I will not add that complexity increase rn

File: src/services/NotificationService.ts (hunk 1 above)
Comment (sg-gs) [reply src/services/NotificationService.ts:38]: we will revisit this in the future

**Review by sg-gs — APPROVED** [review]

**Comment by sg-gs** [comment]
The environment variables ARE NOT still set. Set them before deploying @fabioespinosa or I can do it if required.

**Pull request closed without merging**